src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -119,44 +119,52 @@ private void ValidateWorkloadIdsInput()
-                _reporter.WriteLine(JsonSerializer.Serialize(packageUrls));
-                _reporter.WriteLine("==allPackageLinksJsonOutputEnd==");
-            }
-			else if (!string.IsNullOrWhiteSpace(_downloadToCacheOption))
-            {
-                try
+                    _reporter.WriteLine("==allPackageLinksJsonOutputStart==");
+                    _reporter.WriteLine(JsonSerializer.Serialize(packageUrls));
+                    _reporter.WriteLine("==allPackageLinksJsonOutputEnd==");
+                }
+                else if (!string.IsNullOrWhiteSpace(_downloadToCacheOption))
                 {
-                    DownloadToOfflineCacheAsync(_workloadIds.Select(id => new WorkloadId(id)), new DirectoryPath(_downloadToCacheOption), _skipManifestUpdate, _includePreviews).Wait();
+                    try
+                    {
+                        DownloadToOfflineCacheAsync(_workloadIds.Select(id => new WorkloadId(id)), new DirectoryPath(_downloadToCacheOption), _skipManifestUpdate, _includePreviews).Wait();
+                    }
+                    catch (Exception e)
+                    {
+                        throw new GracefulException(string.Format(LocalizableStrings.WorkloadCacheDownloadFailed, e.Message), e);
+                    }
                 }
-                catch (Exception e)
+                else
                 {
-                    throw new GracefulException(string.Format(LocalizableStrings.WorkloadCacheDownloadFailed, e.Message), e);
+                    try
+                    {
+                        InstallWorkloads(
+                            _workloadIds.Select(id => new WorkloadId(id)),
+                            _skipManifestUpdate,
+                            _includePreviews,
+                            string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption));
+                    }
+                    catch (Exception e)
+                    {
+                        // Don't show entire stack trace
+                        throw new GracefulException(string.Format(LocalizableStrings.WorkloadInstallationFailed, e.Message), e);
+                    }
                 }
             }
-            else
+            catch (Exception)
             {
-                try
-                {
-                    InstallWorkloads(
-                        _workloadIds.Select(id => new WorkloadId(id)),
-                        _skipManifestUpdate,
-                        _includePreviews, 
-                        string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption));
-                }
-                catch (Exception e)
-                {
-                    // Don't show entire stack trace
-                    throw new GracefulException(string.Format(LocalizableStrings.WorkloadInstallationFailed, e.Message), e);
-                }
+                _workloadInstaller.Shutdown();
```

Comment:
How reliable is the shutdown method? If it throws here it will cover up the original installation failure.

Reply:
That is a possibility. I'm open to suggestions. For example, if the user starts a command and hits ctrl+break or starts to logoff. We can handle those, but they're very platform specific.

Reply:
Personally I'm less concerned about those specific scenarios, since the user is getting themselves into a bad state if they take those actions and it can be fixed with repair. I just want to make sure that we don't have issues where our bug in shutdown covers up real install errors